toolchain/lowering/lowering_handle.cpp

```diff
@@ -24,16 +24,39 @@ auto LoweringHandleAddressOf(LoweringFunctionContext& context,
   context.SetLocal(node_id, context.GetLocal(node.GetAsAddressOf()));
 }
 
-auto LoweringHandleArrayType(LoweringFunctionContext& /*context*/,
-                             SemanticsNodeId /*node_id*/, SemanticsNode node)
+auto LoweringHandleArrayValue(LoweringFunctionContext& context,
+                              SemanticsNodeId node_id, SemanticsNode node)
     -> void {
-  CARBON_FATAL() << "TODO: Add support: " << node;
-}
+  auto* llvm_type = context.GetType(node.type_id());
+  auto* alloca =
+      context.builder().CreateAlloca(llvm_type, /*ArraySize=*/nullptr, "array");
+  context.SetLocal(node_id, alloca);
+  auto array_size = llvm_type->getArrayNumElements();
+  auto tuple_node_id = node.GetAsArrayValue();
+  auto* tuple_value = context.GetLocal(tuple_node_id);
+  auto* tuple_type =
+      context.GetType(context.semantics_ir().GetNode(tuple_node_id).type_id());
 
-auto LoweringHandleArrayValue(LoweringFunctionContext& /*context*/,
-                              SemanticsNodeId /*node_id*/, SemanticsNode node)
-    -> void {
-  CARBON_FATAL() << "TODO: Add support: " << node;
+  // For return values or call instructions, the tuple is not in the memory so
+  // it cannot be accessed directly to create the array value. Hence, we are
+  // creating a store in this block.
+  if (!tuple_value->getType()->isPointerTy()) {
+    tuple_value = context.builder().CreateAlloca(
+        tuple_type, /*ArraySize=*/nullptr, "as_tuple");
+    context.builder().CreateStore(context.GetLocalLoaded(tuple_node_id),
+                                  tuple_value);
+  }
+
+  for (uint64_t i = 0; i < array_size; i++) {
```

Comment:
```suggestion
  for (int i = 0; i < llvm_type->getArrayNumElements(); ++i) {
```

There are a few stylistic things here:

- We tend to prefer int: https://google.github.io/styleguide/cppguide.html#Integer_Types
- We prefer variables to be close to their usage, but array_size is only used here: https://google.github.io/styleguide/cppguide.html#Local_Variables
- We prefer prefix increment: https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement